src/onyx/log/commands/prepare_join_cluster.clj

```diff
@@ -93,10 +93,25 @@
             :args {:observer (:subject diff)}}])))
 
 (s/defmethod extensions/fire-side-effects! :prepare-join-cluster :- State
-  [{:keys [args message-id]} :- LogEntry old new diff {:keys [monitoring] :as state}]
+  [{:keys [args message-id]} :- LogEntry old new diff {:keys [log monitoring] :as state}]
   (common/start-new-lifecycle
    old new diff
-   (cond (= (:id state) (:observer diff))
+   (cond ;; Handles the cases where all peers are actually dead.
+         ;; This can happen if a single node cluster comes down
+         ;; and is rebooted. We pick a predictably-random peer
+         ;; and knock it down if it's not up. This garuntees
```

Comment:
Thank you :D

On 11 February 2016 at 23:01, Szabó Krisztián <notifications@github.com>
wrote:

> In src/onyx/log/commands/prepare_join_cluster.clj
> <https://github.com/onyx-platform/onyx/pull/526#discussion_r52611929>:
>
> >    (common/start-new-lifecycle
> >     old new diff
> > -   (cond (= (:id state) (:observer diff))
> > +   (cond ;; Handles the cases where all peers are actually dead.
> > +         ;; This can happen if a single node cluster comes down
> > +         ;; and is rebooted. We pick a predictably-random peer
> > +         ;; and knock it down if it's not up. This garuntees
>
> s/garuntees/guarantees
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/onyx-platform/onyx/pull/526/files#r52611929>.
>